middle_end/flambda2/terms/flambda_primitive.ml

```diff
@@ -20,6 +20,12 @@
 
 module K = Flambda_kind
 
+let fprintf = Format.fprintf
+
+let ( let* ) c f = if c <> 0 then c else f ()
+
+let constructor_tag p = Obj.tag (Obj.repr p)
```

Comment:
I don't think this is a good idea: sure it may reduce the number of lines a bit, but 1) as pointed out by @Ekdohibs later, it is incorrect on nullary constructors (where it will return the same values for all nullary constructors) and 2) it's slow since it's actually an external call.